### ISSUE TITLE
[Feature]: Upgrade as many vLLM built-in logits processors to the new logits processors programming model as possible

### ISSUE THREAD

**Issue by afeldman-nm** [vllm-project/vllm#25957]
### 🚀 The feature, motivation and pitch

vLLM v1 is intended to handle logits processors differently than vLLM v0. Logits processors should subclass `LogitsProcessor`; built-in logits processors should be implemented in `vllm/v1/sample/logits_processor/builtin.py` and added to the `BUILTIN_LOGITS_PROCESSORS` list in `vllm/v1/sample/logits_processor/__init__.py`

However, in actuality only several of the vLLM built-in logits processors (Min-P, logits bias, min tokens) have been converted to work this way; the rest are still hard-coded into the engine core and are not implemented as `LogitsProcessor` subclasses. The reason for this is that there is an open issue to revise the logits processor programming model ( #25389 ) and it makes sense to resolve this issue before converting these hard-coded logits processors to the new style.

Once #25389 is resolved, it makes sense to consider which of the following hard-coded logits processors can be updated to the new style:

* Allowed token IDs
* Bad words
* Repetition penalty
* Frequency penalty
* Presence penalty
* Temperature
* Top-K
* Top-P

### Alternatives

Until #25389 is resolved, incremental improvements can be made to the existing hard-coded logits processors without upgrading them to be subclasses of `LogitsProcessors`, exemplified by #19482 .

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.
